### ISSUE TITLE
Post order returned from cursors is not stable

### ISSUE THREAD

**Issue by epeli** [wp-graphql/wp-graphql#729]
When paginating data using Apollo I randomly get same items from different cursor pages

![image](https://user-images.githubusercontent.com/225712/54748196-a2090f80-4bd9-11e9-9348-a9085fa7c366.png)

I cannot isolate this to a test because this happens randomly. I first noticed this while implementing meta value based ordering in #721 but today I noticed it happening without meta values too so I'm opening this new issue.

I'm not using custom ordering here but I'm using wp-graphql-tax-query based filtering.

Luckily my refactoring in #721 fixes this issue too. It changes the generated SQL query from

```sql
SELECT   wp_posts.ID FROM wp_posts  LEFT JOIN wp_term_relationships ON (wp_posts.ID = wp_term_relationships.object_id) WHERE 1=1  AND wp_posts.post_parent = 0  AND ( 
  ( 
    wp_term_relationships.term_taxonomy_id IN (23)
  )
) AND wp_posts.post_type = 'apartment' AND ((wp_posts.post_status = 'publish')) AND wp_posts.post_date <= '2019-03-18 19:04:34' AND wp_posts.ID != 481 GROUP BY wp_posts.ID ORDER BY wp_posts.post_date DESC LIMIT 0, 6
```

to

```sql
SELECT   wp_posts.ID FROM wp_posts  LEFT JOIN wp_term_relationships ON (wp_posts.ID = wp_term_relationships.object_id) WHERE 1=1  AND wp_posts.post_parent = 0  AND ( 
  ( 
    wp_term_relationships.term_taxonomy_id IN (23)
  )
) AND wp_posts.post_type = 'apartment' AND ((wp_posts.post_status = 'publish')) AND wp_posts.ID < 461 GROUP BY wp_posts.ID ORDER BY wp_posts.post_date DESC, wp_posts.ID DESC  LIMIT 0, 6
```

The post_date AND  `AND wp_posts.post_date <= '2019-03-18 19:04:30'` is gone and `, wp_posts.ID DESC` is added to the `GROUP BY` clause.

100% sure this is the best way to do but seems to fix my app. I can now get all values using cursors without duplicates.

**Comment by epeli** [comment]
Ah, now I understand why the current implementation is broken.

In the cursor implementation the `AND` operators must match with the `GROUP BY` clauses and there's is no `post_date` based ordering in the `GROUP BY`.

The current implementation defaults to filtering by `post_date` but WordPress does not do it so there's a miss match. My new implementation just doesn't happen to add `post_date` AND operator in this case so it works. But it tries to do it... I need to remove it completely.

**Comment by epeli** [comment]
Made sure that #721 now never compares by dates (unless explicitly asked).

**Comment by jasonbahl** [comment]
I need to read up on this, but dates in WP are weird too because any number of posts can have the same date, down to the second. WP doesn’t force unique dates at any level, so funkiness can ensue when the cursor post date is identical to another posts date. 

May or may not be relevant. I need to read though what you’ve worked on still. I’ll be at a computer in a bit over an hour.

**Comment by epeli** [comment]
Hah, now I know why this bug appears in my current project. I have bunch of data imported automatically to a post type and many of them have the same date timestamp. So MySQL ends up comparing the same date strings and result of that seems to be bit random.

**Comment by epeli** [comment]
No wait. There's actually is date in the original `GROUP BY`... I need to investigate more.

**Comment by epeli** [comment]
I think I got to bottom of this https://github.com/wp-graphql/wp-graphql/pull/730

**Comment by jasonbahl** [comment]
Closed by #744 

@epeli if you come across further issues, let's open a new issue and reference this one if necessary
